src/components/ChangePlan/CheckoutProcess.js

```diff
@@ -0,0 +1,361 @@
+                  <p>{_('checkoutProcessForm.success_msg')}</p>
+                </div>
+              </div>
+            ) : null}
+            <div className="dp-footer-form" style={flextStartSyle}>
+              <div className="dp-action">
+                <SubmitButton className="dp-button button-medium primary-green">
+                  {_('checkoutProcessForm.save_continue')}
+                </SubmitButton>
+              </div>
+            </div>
+          </fieldset>
+        </FormWithCaptcha>
+      ) : null}
+    </div>
+  );
+};
+
+const CheckoutProcess = ({
+  location,
+  dependencies: { planService, dopplerUserApiClient, appSessionRef },
+}) => {
+  const intl = useIntl();
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
+  const { pathType, planType } = useRouteMatch().params;
+
+  const [steps, setSteps] = useState({
+    contactInformation: true,
+    billingInformation: false,
+    paymentInformation: false,
+  });
+
+  const submitContactInformationForm = async (values) => {
+    const result = await dopplerUserApiClient.createOrUpdateContactInformation(values);
+
+    if (result.success) {
+      setActiveStep();
+    }
+
+    return result;
+  };
+
+  const setActiveStep = (target) => {
+    switch (target) {
+      case 'contact-information':
+        setSteps({
+          contactInformation: true,
+          billingInformation: false,
+          paymentInformation: false,
+        });
+        break;
+      case 'billing-information':
+        setSteps({
+          contactInformation: false,
+          billingInformation: true,
+          paymentInformation: false,
+        });
+        break;
+      case 'payment-information':
+        setSteps({
+          contactInformation: false,
+          billingInformation: false,
+          paymentInformation: true,
+        });
+        break;
+    }
+  };
+
+  const [state, setState] = useState({ loading: true });
+
+  useEffect(() => {
+    const mapCountry = (country) => {
+      return {
+        key: country.id,
+        value: country.name,
+      };
+    };
+
+    const fetchData = async () => {
+      const contactInformationResult = await dopplerUserApiClient.getContactInformationData();
+      const countriesResult = await dopplerUserApiClient.getCountriesData();
+
+      const countries = countriesResult.success ? countriesResult.value.map(mapCountry) : [];
+      countries.splice(0, 0, { key: '', value: _('checkoutProcessForm.empty_option_select') });
+
+      setState({
+        contactInformation: contactInformationResult.success
+          ? contactInformationResult.value
+          : null,
+        loading: false,
+        countries: countries,
+      });
+
+      setSteps({ contactInformation: true, billingInformation: false, paymentInformation: false });
+    };
+    fetchData();
+  }, [appSessionRef, planService, planType, pathType]);
+
+  if (state.loading) {
+    return <Loading page />;
+  }
```

Comment:
Why not adding this inside the rest of the jsx code in line 330 like we do [here](https://github.com/FromDoppler/doppler-webapp/blob/master/src/components/Reports/Reports.js#L116)

Reply:
yes, I will change it.